test/wasi/test.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+require('../common');
+
+const cp = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+const SKIP = [
+  'poll',
```

Comment:
if anyone is interesting in these tests:
https://github.com/devsnek/node-wasi/issues/3
https://github.com/devsnek/node-wasi/issues/4
https://github.com/devsnek/node-wasi/issues/5

Reply:
I believe the `write_file` test is failing because the file isn't being opened for writing correctly. The test's C code is opening the file in `'w'` mode. In your `convertOpenFlags()` function, the mapping is incomplete. This is what `openSync()` expects: https://github.com/nodejs/node/blob/5b8df5e956680dc1a38b631e53f5e70a905fd917/lib/internal/fs/utils.js#L337

Reply:
fantastic catch, thank you so much! stat is still failing but not on trying to write anymore 👍

Reply:
It looks like the stat test may need to handle `mkdir()` failures if the directory already exists, and `SIZE` is 500, but only 1 byte is written.

Reply:
well its supposed to seek out to 499, but libuv doesn't have seeking, so i'm working locally rn on adding position state to fake seeking.

Reply:
It might also be time for libuv to learn some seeking APIs. I know I've seen at least one previous request for it.

Reply:
according to old issues they aren't safe with libuv's concurrency. perhaps they could be allowed for sync calls though?